src/common/src/array/arrow.rs

```diff
@@ -17,9 +17,37 @@ use arrow_schema::Field;
 use chrono::{NaiveDateTime, NaiveTime};
 use itertools::Itertools;
 
+use super::column::Column;
 use super::*;
 use crate::types::struct_type::StructType;
 
+// Implement bi-directional `From` between `DataChunk` and `arrow_array::RecordBatch`.
+
+impl From<&DataChunk> for arrow_array::RecordBatch {
+    fn from(chunk: &DataChunk) -> Self {
+        arrow_array::RecordBatch::try_from_iter(
+            chunk
+                .columns()
+                .iter()
+                .map(|column| ("", column.array_ref().into())),
+        )
+        .unwrap()
+    }
+}
+
+impl From<&arrow_array::RecordBatch> for DataChunk {
+    fn from(batch: &arrow_array::RecordBatch) -> Self {
+        DataChunk::new(
+            batch
+                .columns()
+                .iter()
+                .map(|array| Column::new(Arc::new(array.into())))
+                .collect(),
+            batch.num_rows(),
```

Comment:
This is to say, we will never have invisible rows in the arrow format?

Reply:
Yeah? Seems that arrow `RecordBatch` doesn't have the concept of visibility like us. So we may need to compact our data chunk before sending it out.

Reply:
fixed in f4c53748